lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -892,6 +902,25 @@ instance Buildable TxMeta where
         <> " " <> build s
         <> " since " <> build sl <> "#" <> build bh
 
+-- | Semi-structured application-specific extension data which is embedded
+-- in every transaction on chain.
+--
+-- See Appendix E of the delegation and incentives design spec.
+--
+-- fixme: TxMeta / TxMetadata naming is rather confusing.
+newtype TxMetadata = TxMetadata { unTxMetadata :: Value }
+  deriving (Show, Eq, Generic)
+
+instance NFData TxMetadata
+
+-- Compare metadatas by json encoding
+instance Ord TxMetadata where
+    compare = comparing (Aeson.encode . unTxMetadata)
```

Comment:
What's the rationale here ? Why not rely on the existing `Ord` instance for `Value` ?

Reply:
`Value` has no `Ord` ... nor does `Cardano.Api.TxMetadata`.
I define it by string encoding so that other types can keep their `Ord` constraint.